app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ReportingRepository.kt

```diff
@@ -132,22 +141,95 @@ class ReportingRepositoryImpl(
         tanUuid = apiInteractor.requestTan(mobileNumber).uuid
     }
 
-    override suspend fun uploadReportInformation(
-        temporaryExposureKeys: List<TemporaryExposureKey>
-    ): MessageType {
+    override suspend fun uploadReportInformation(): MessageType {
         return when (messageTypeSubject.value) {
-            MessageType.Revoke.Suspicion -> uploadRevokeSuspicionInfo(temporaryExposureKeys)
-            MessageType.Revoke.Sickness -> uploadRevokeSicknessInfo(temporaryExposureKeys)
-            else -> uploadInfectionInfo(temporaryExposureKeys)
+            MessageType.Revoke.Suspicion -> uploadRevokeSuspicionInfo()
+            MessageType.Revoke.Sickness -> uploadRevokeSicknessInfo()
+            else -> uploadInfectionInfo()
         }
     }
 
-    private suspend fun uploadInfectionInfo(
-        temporaryExposureKeys: List<TemporaryExposureKey>): MessageType.InfectionLevel {
+    private suspend fun uploadInfectionInfo(): MessageType.InfectionLevel {
         return withContext(coroutineContext) {
-            val infectionLevel = messageTypeSubject.value as? MessageType.InfectionLevel ?: throw InvalidConfigurationException.InfectionLevelNotSet
+            val infectionLevel = messageTypeSubject.value as? MessageType.InfectionLevel
+                ?: throw InvalidConfigurationException.InfectionLevelNotSet
+
+            val configuration = configurationRepository.observeConfiguration().blockingFirst()
+            val uploadKeysDays = configuration.uploadKeysDays
+                ?: throw InvalidConfigurationException.NullNumberOfDaysToUpload
+            var thresholdTime = ZonedDateTime.now()
+                .minusDays(uploadKeysDays.toLong())
+                .startOfTheDay()
```

Comment:
here is the start of the Day. I belive we should to the same on top.

Reply:
I would say in this case the use of `startOfTheDay` is valid because we are working with days. We need to look up two days in the past. 
In the previous example the `yellowWarningQuarantine` and `yellowWarningQuarantine` are in hours.